examples/cursor_sync_custom_overlay.py

```diff
@@ -0,0 +1,106 @@
+"""
+Cursor sync via custom overlay
+==============================
+
+Create 2 seperate viewers and mirror the position of the cursor between the
+parent and the child via a vispy overlay.
+
+Read up on the view model here: https://napari.org/stable/guides/napari_models.html
+
+.. tags:: visualization-basic, gui
+"""
+
+import warnings
+
+import numpy as np
+from vispy.scene.visuals import Ellipse
+
+import napari
+from napari._vispy.overlays.base import ViewerOverlayMixin, VispyCanvasOverlay
+from napari._vispy.utils.visual import overlay_to_visual
+from napari.components.overlays import CanvasOverlay
+from napari.utils.events import Event
+
+#
+#   Create a cursor overlay
+#
+
+#   NOTE:
+#   Read up on the view model to understand the reasons for this:
+#   https://napari.org/stable/guides/napari_models.html
+
+
+class CursorOverlay(CanvasOverlay):
+    """Used as the key for overlay_to_visual's overlay dict"""
+
+
+class VispyCursorOverlay(ViewerOverlayMixin, VispyCanvasOverlay):
+
+    def __init__(self, *, viewer, overlay, parent=None):
+
+        cursor = Ellipse(center=(4, 4), color='red', radius=4)
+
+        super().__init__(
+            node=cursor,
+            viewer=viewer,
+            overlay=overlay,
+            parent=parent,
+        )
+
+        self.reset()
+
+    def reset(self):
+        super().reset()
+
+
+#
+#   Register the new "cursor" in the global `overlay_to_visual`
+#
+
+overlay_to_visual[CursorOverlay] = VispyCursorOverlay  # type: ignore
+
+
+#
+#   Wire it up
+#
+
+def sync_cursor(parent: napari.Viewer, child: napari.Viewer):
+    """Syncs a 'cursor' onto the child, relative to the parent's canvas"""
+
+    cursor_overlay = CursorOverlay(visible=True)  # create a new "cursor" overlay
+    child._overlays['t_cursor'] = cursor_overlay  # add to the child
+    child.window._qt_viewer.canvas._add_overlay_to_visual(cursor_overlay)  # trigger overlay generation
+
+    @parent.cursor.events.position.connect
+    def sync_cursor_parent_callback(e: Event):
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore')
+
+            t_cursor: CursorOverlay = child._overlays['t_cursor']
+            cursor: VispyCursorOverlay = child.window._qt_viewer.canvas._overlay_to_visual[t_cursor]
+
+            # Get position of real cursor relative to the parent's canvas
+            parent_global_pos = parent.window._qt_viewer.cursor().pos()
+            parent_local_pos = parent.window._qt_viewer.mapFromGlobal(parent_global_pos)
+
+            child_x = parent_local_pos.x()
+            child_y = parent_local_pos.y()
+
+            # Apply new position to "cursor"
+            transform = [child_x, child_y, 0, 0]
+            cursor.node.transform.translate = transform
```

Comment:
I find it a little weird that we modify the translate parameter of the ellipse rather than changing the center. Is it weird or is it just me? 😅